packages/bot/src/scratch/utils.js

```diff
@@ -11,21 +10,22 @@ export const oppositesToDropdownOptions = opposite_name => {
     });
 };
 
-export const cleanUpOnLoad = (blocksToClean, dropEvent) => {
-    const { clientX = 0, clientY = 0 } = dropEvent || {};
-    const blocklyMetrics = Blockly.mainWorkspace.getMetrics();
-    const scaleCancellation = 1 / Blockly.mainWorkspace.scale;
-    const blocklyLeft = blocklyMetrics.absoluteLeft - blocklyMetrics.viewLeft;
-    const blocklyTop = document.body.offsetHeight - blocklyMetrics.viewHeight - blocklyMetrics.viewTop;
-    const cursorX = clientX ? (clientX - blocklyLeft) * scaleCancellation : 0;
-    let cursorY = clientY ? (clientY - blocklyTop) * scaleCancellation : 0;
-    blocksToClean.forEach(block => {
-        block.moveBy(cursorX, cursorY);
+export const cleanUpOnLoad = (blocks_to_clean, drop_event) => {
+    const { clientX = 0, clientY = 0 } = drop_event || {};
+    const blockly_metrics = Blockly.derivWorkspace.getMetrics();
+    const scale_cancellation = 1 / Blockly.derivWorkspace.scale;
```

Comment:
i think we need a comment on what is happening here 😂

Reply:
I don't have time look back at the code.😂  All you need to know is all the blocks rearrange after script is loaded